src/main/java/cz/centrum/haffner/SimpleTrainingDavid/AppServices/CountryCodeExtractor.java

```diff
@@ -5,7 +5,7 @@
 import org.springframework.stereotype.Component;
 
 @Component
-public class CountryCodeExtractor {
+public class CountryCodeExtractor implements CountryCodeExtractorInt {
```

Comment:
Co se vytváření interfaců týče, tak se mrkni na tohle:
https://stackoverflow.com/questions/2814805/java-interfaces-implementation-naming-convention
Jde vidět, že různí lidé mají různé názory na to, jak to dělat správně.
Já osobně se kloním k tomu že pokud se píše Java, tak by se měly používat konvence Javy co se vytváření interface a jejich implementací týče. (Příklad: Interface = "List", Implementace = "ArrayList", "LinkedList" atd...)
Rozhodně bych název Interface nevytvářel přidáním suffixu "Int", vzhledem k tomu, že to spíš evokuje Integer.
Prefix "I" mi příjde schůdný.
Takže by dávalo smysl, pojmenovat si interface třeba jako Extractor (nebo IExtractor) s jednou metodou 
extract(Object objectForExtraction)
a potom jeho implementace by byla CountryCodeExtractor s konkrétní implementací extrakce.

Reply:
Pak z toho už jde do budoucna cítit, že pokud by jiný developer potřeboval přidat nějakou další extrakci, která vrací int, tak by implementoval interface Extractor.
Případně by rozšířil interface Extractor o metodu extract(), který by vracela typ, který potřebuje on (třeba String) a tu by si implementoval.
S tím, že do interface Extractor by vytvořil její defaultní implementaci (klidně vracející null), aby se nemusela implementovat ve všech ostatních implementacích tohoto interfacu.